### PR TITLE
elantp: Modify firmware update flow in bootloader mode

### PR DESCRIPTION
**Pull request by jinglewu**
Modify the firmware bin file size.
Modify firmware update flow when the firmware in bootloader mode.

### CODE REVIEW

**Comment by hughsie** [comment]
Can you prefix all the commit titles with `elantp: ` please.

File: plugins/elantp/fu-elantp-hid-device.c [hunk 1]
```diff
@@ -254,7 +224,7 @@ fu_elantp_hid_device_setup (FuDevice *device, GError **error)
 			     ic_type);
 		return FALSE;
 	}
-	fu_device_set_firmware_size (device, (guint64) self->ic_page_count * (guint64) self->fw_page_size);
+	fu_device_set_firmware_size (device, (guint64) self->ic_page_count * (guint64) 64);
```
Comment (hughsie) [inline plugins/elantp/fu-elantp-hid-device.c:227]: I think we need a comment above explaining why it's being hardcoded here.

File: plugins/elantp/fu-elantp-hid-device.c [hunk 2]
```diff
@@ -377,21 +347,81 @@ static gboolean
...
 	/* sanity check */
 	if (fu_device_has_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER)) {
-		g_debug ("already in bootloader mode, skipping");
-		return TRUE;
-	}
-
-	g_debug ("in bootloader mode, reset IC");
-	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
-	if (!fu_elantp_hid_device_write_cmd (self,
+		//g_debug ("already in bootloader mode, skipping");
```
Comment (hughsie) [inline plugins/elantp/fu-elantp-hid-device.c:356]: no C++ comments please

File: plugins/elantp/fu-elantp-hid-device.c [hunk 3]
```diff
@@ -377,21 +347,81 @@ static gboolean
 fu_elantp_hid_device_detach (FuDevice *device, GError **error)
 {
 	FuElantpHidDevice *self = FU_ELANTP_HID_DEVICE (device);
-
+	guint16 iap_ver;
+	guint16 ic_type;
+	guint8 buf[2] = { 0x0 };
+	guint16 tmp;
 	/* sanity check */
```
Comment (hughsie) [inline plugins/elantp/fu-elantp-hid-device.c:354]: needs a newline between the C99 variable block and the `/*` comment

File: plugins/elantp/fu-elantp-hid-device.c [hunk 4]
```diff
@@ -162,6 +163,8 @@ fu_elantp_hid_device_setup (FuDevice *device, GError **error)
 		return FALSE;
 	}
 	fwver = fu_common_read_uint16 (buf, G_LITTLE_ENDIAN);
+	if (fwver==0xFFFF || fwver==ETP_CMD_I2C_FW_VERSION)
```
Comment (hughsie) [inline plugins/elantp/fu-elantp-hid-device.c:166]: space either size of the `==` please

**Comment by hughsie** [comment]
Looks good. Can you `git rebase master -i` and squash/fixup these down into less commits please? Then looks good to me.

**Comment by hughsie** [comment]
@jinglewu there's still 9 commits on this PR...

**Comment by superm1** [comment]
> @jinglewu there's still 9 commits on this PR...
@jinglewu We're getting close to cutting 1.5.0, and if you want this included it should be merged pretty soon.

Do you need some guidance on rebasing?  If so, you may take a look here for how to do it:
https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

**Comment by jinglewu** [comment]
> > @jinglewu there's still 9 commits on this PR...
> 
> @jinglewu We're getting close to cutting 1.5.0, and if you want this included it should be merged pretty soon.
> 
> Do you need some guidance on rebasing? If so, you may take a look here for how to do it:
> https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

**Comment by jinglewu** [comment]
> > > @jinglewu there's still 9 commits on this PR...
> > 
> > 
> > @jinglewu We're getting close to cutting 1.5.0, and if you want this included it should be merged pretty soon.
> > Do you need some guidance on rebasing? If so, you may take a look here for how to do it:
> > https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

so 1.5.0 will contain these patches?

**Comment by superm1** [comment]
If you can rebase, squash and address any concerns, yes it can be included in upcoming 1.5.0 release.

**Comment by jinglewu** [comment]
> If you can rebase, squash and address any concerns, yes it can be included in upcoming 1.5.0 release.
Do i want to rebase "jinglewu:master merge to fwupd:master" ?

**Comment by superm1** [comment]
Correct

**Comment by jinglewu** [comment]
> Correct

https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec

->   This website is an example “rebase 9 commit to 1 commit in my  "jinglewu:master”
->	How to merge "jinglewu:master to fwupd:master" ?

**Comment by superm1** [comment]
I would start to rebase from your local branch and squash those commits, and then after they're squashed can rebase on fwupd:master.

**Comment by jinglewu** [comment]
> I would start to rebase from your local branch and squash those commits, and then after they're squashed can rebase on fwupd:master.

so Currently, I don't need to rebase this 9 commits to 1 commit now?

**Comment by superm1** [comment]
Please squash the commits to one with rebase command

**Comment by jinglewu** [comment]
> Please squash the commits to one with rebase command
I have successfully rebase, what I need to do next?

**Comment by superm1** [comment]
Push back to your GitHub with --force argument

**Pull request merged**